src/webview/js/fixup-katex.js

```diff
@@ -0,0 +1,35 @@
+/* @flow strict */
+
+// Auxiliary function, for brevity.
+const makeDiv = (className: string): Element => {
+  const element = document.createElement('div');
+  element.classList.add(className);
+  return element;
+};
+
+/**
+ * Adjust non-inline KaTeX equation element structure.
+ *
+ * This surrounds existing `.katex-display` elements (the KaTeX wrapper for
+ * non-inline equations) with shim `<div>`s to allow horizontal scrolling.
+ *
+ * (The actual functionality of these elements is provided by CSS rules.)
+ */
+const fixupKatex = (root: Element) => {
+  Array.from(root.querySelectorAll('.katex-display')).forEach(kd => {
+    // nodes returned by `querySelectorAll` will always have a valid parent node
```

Comment:
This is true in general! `root.querySelectorAll(...)` never returns `root` as one of its values.

(It looks like it also doesn't return nodes from shadow DOM subtrees, although I admit I hadn't considered that at the time.)

Reply:
I see. Because it only returns descendants, and "descendant" [is defined](https://dom.spec.whatwg.org/#concept-tree-descendant) to exclude the ancestor itself.

As always, when I ask a question in code review because I'm confused, I would like the confusion cleared up in the branch itself. :-)